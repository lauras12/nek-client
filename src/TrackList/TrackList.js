import React from 'react';
import TrailContext from '../Context';
import TrackListItem from '../TrackListItem/TrackListItem';
import APITrackCalls from '../services/API_Track_service';
import './TrackList.css';

export default class TrackList extends React.Component {
    static contextType = TrailContext;
    state = {
        error: null,
    }

    componentDidMount = () => {
        APITrackCalls.getAllTracksData()
            .then(data => {
               // console.log(data);
                this.context.setTracksList(data);
            })
            .catch(res => {
                this.setState({
                    error: res
                });
            });
    }

    render() {

        const tracks = this.context.tracks.map(track => {
            return <TrackListItem
                key={track.id}
                id={track.id}
                name={track.alias}
                img={track.img}
            />;
            
        });
        // console.log(tracks);
        return (
            <div className='track-list'>
                <div className='error'>
                    {this.state.error ? this.state.error.message : null}
                </div>
                <h2 className='title'>TRACKS LIBRARY: </h2>
                <ul className='tracks-container'>
                    {tracks}
                </ul>
            </div>
        );
    }
}