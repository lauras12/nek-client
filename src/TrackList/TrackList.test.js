import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import TrackList from './TrackList';
import {BrowserRouter as BR } from 'react-router-dom';


describe('TrackList component', () => {
    
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><TrackList /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const list = renderer.create(<BR><TrackList /></BR>);
        expect(list.toJSON()).toMatchSnapshot();
    });
});
