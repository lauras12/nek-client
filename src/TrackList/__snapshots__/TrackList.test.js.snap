// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList component renders UI as expected 1`] = `
<div
  className="track-list"
>
  <div
    className="error"
  />
  <h2
    className="title"
  >
    TRACKS LIBRARY: 
  </h2>
  <ul
    className="tracks-container"
  />
</div>
`;
